torch/_prims/utils.py

```diff
@@ -0,0 +1,468 @@
+import torch
+
+from numbers import Number
+from typing import Union, Sequence
+from functools import reduce
+
+
+class TensorMeta(object):
+    """
+    Temporary helper class to model tensor metadata.
+
+    To be replaced with an actual meta tensor.
```

Comment:
It doesn't seem to be quite equivalent to meta tensor though, as it doesn't model storage/storage offset.